### ISSUE TITLE
Belajar GAN

### ISSUE THREAD

**Issue by threeal** [threeal-archive/research#22]
Repositori di [sini](https://github.com/threeal/Introduction-to-GAN), Report di [sini](https://docs.google.com/document/d/10NpWZ-RL8wrddN9PbWbcZXvgQqBPjzwFo73VsWdz8YA/edit?usp=sharing).
- [x] Fundamental of GANs, [here](https://github.com/threeal/Introduction-to-GAN/tree/master/1%20-%20Fundamental%20of%20GANs).
  - [x] Slide, [here](https://docs.google.com/presentation/d/1CR8LCBkkIiygs3xAwd-GewdYvo3TfeasU8wJSohLLu8/edit?usp=sharing).
  - [x] GAN, [GitHub](https://github.com/threeal/Introduction-to-GAN/blob/master/1%20-%20Fundamental%20of%20GANs/GAN.ipynb), [Colab](https://colab.research.google.com/drive/1L8rG7tPMLFRgc_HpIJNHkJcLvHB51XQw).
  - [x] CGAN, [GitHub](https://github.com/threeal/Introduction-to-GAN/blob/master/1%20-%20Fundamental%20of%20GANs/CGAN.ipynb), [Colab](https://colab.research.google.com/drive/1EykgqqzCJQpDvuq33GVf3fimULoQQEWO).
  - [x] DCGAN, [Tutorial](https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html), [GitHub](https://github.com/threeal/Introduction-to-GAN/blob/master/1%20-%20Fundamental%20of%20GANs/DCGAN.ipynb).
- [ ] Wasserstein GANs, [here](https://github.com/threeal/Introduction-to-GAN/tree/master/2%20-%20Wasserstein%20GANs).
  - [x] Slide, [here](https://docs.google.com/presentation/d/15-y36v9MACwwUmJ4XMR-6n5JW40Pcwd5oAqWfhZ_6LY/edit?usp=sharing).
  - [x] WGAN, [GitHub](https://github.com/threeal/Introduction-to-GAN/blob/master/2%20-%20Wasserstein%20GANs/WGAN.ipynb).
  - [x] WGAN-GP, [GitHub](https://github.com/threeal/Introduction-to-GAN/blob/master/2%20-%20Wasserstein%20GANs/WGAN-GP.ipynb).
  - [ ] WGAN-SN, [GitHub](https://github.com/threeal/Introduction-to-GAN/blob/master/2%20-%20Wasserstein%20GANs/WGAN-SN.ipynb).
  - [ ] WGAN-Div, [GitHub](https://github.com/threeal/Introduction-to-GAN/blob/master/2%20-%20Wasserstein%20GANs/WGAN-Div.ipynb).
- [ ] GAN Evaluations, [here](https://github.com/threeal/Introduction-to-GAN/tree/master/3%20-%20GAN%20Evaluations).
  - [ ] Slide, [here](https://docs.google.com/presentation/d/1Z5OR-8cEIfVcfyoFYHrC7haPHWq1hDuZOqyL-MYzadk/edit?usp=sharing).
- [ ] Applications of GANs, [here](https://github.com/threeal/Introduction-to-GAN/tree/master/4%20-%20Applications%20of%20GANs).

## Things to Know

- [ ] Generative Adversarial Networks (GAN).
  - [x] Minimax GAN (MM GAN).
  - [x] Non-Saturating GAN (NS GAN).
  - [x] Least Squares GAN (LS GAN).
  - [x] Deep Convolutional GAN (DCGAN).
  - [x] Conditional GAN (CGAN).
  - [x] Wasserstein GAN (WGAN).
    - [ ] Wasserstein GAN with Gradient Penalty (WGAN-GP).
    - [ ] Wasserstein Divergence GAN (WGAN-Div).
  - [ ] Spectral Normalization GAN (SN-GAN).
- [x] Multi-layer Perceptron (MLP).
- [ ] Deep Feed-forward Neural Networks (DNN).
- [x] Convolution Neural Networks (CNN).
- [ ] Recurrent Neural Networks (RNN).
  - [ ] Long Short Term Memory (LSTM).
- [ ] Mean Squared Error (MSE).
- [ ] Binary Cross-Entropy (BCE).
- [ ] Stochastic Gradient Descent (SGD).
- [ ] Adam Optimization.
- [ ] RMSprop.
- [ ] Nash Equilibrium.
- [ ] Earth Mover's Distance.
- [ ] Lipschitz Constraints / Continuity

## Papers

- [x] Goodfellow 2014, _Generative Adversarial Networks_, [arXiv](https://arxiv.org/abs/1406.2661).
- [x] Mirza 2014, _Conditional Generative Adversarial Nets_, [arXiv](https://arxiv.org/abs/1411.1784).
- [x] Radford 2015, _Unsupervised Representation Learning with Deep Convolutional Generative Adversarial Networks_, [arXiv](https://arxiv.org/abs/1511.06434).
- [ ] Arjovsky 2017, _Wasserstein GAN_, [arXiv](https://arxiv.org/abs/1701.07875).
- [ ] Gulrajani 2017, _Improved Training of Wasserstein GANs_, [arXiv](https://arxiv.org/abs/1704.00028).
- [ ] Miyato 2018, _Spectral Normalization for Generative Adversarial Networks_, [Open Review](https://openreview.net/forum?id=B1QRgziT-).
- [ ] Wu 2018, _Wasserstein Divergence for GANs_, [arXiv](https://arxiv.org/abs/1712.01026).

## GAN Related Research

- [x] Heng 2017, _Transfer Learning for Robot Control with Generative Adversarial Networks_, [PDF](https://www.doc.ic.ac.uk/~ejohns/Documents/jonathan_heng_thesis.pdf).
- [ ] Tobin 2017, _Domain randomization for transferring deep neural networks from simulation to the real world_, [IEEE Xplore](https://ieeexplore.ieee.org/abstract/document/8202133),

## Other References

- [x] A gentle introduction to GAN [here](https://machinelearningmastery.com/what-are-generative-adversarial-networks-gans/).
- [x] How do CNN work [here](https://machinelearningmastery.com/convolutional-layers-for-deep-learning-neural-networks/).
- [x] A gentle introduction to ReLU [here](https://machinelearningmastery.com/rectified-linear-activation-function-for-deep-learning-neural-networks/).
- [x] Neural networks representation [here](https://www.jeremyjordan.me/intro-to-neural-networks/).
- [x] Convolution neural networks [here](https://www.jeremyjordan.me/convolutional-neural-networks/).
- [x] Introduction to RNN [here](https://www.jeremyjordan.me/introduction-to-recurrent-neural-networks/).
- [x] How Back-Propragation in ANN work [here](https://towardsdatascience.com/how-does-back-propagation-in-artificial-neural-networks-work-c7cad873ea7).
- [ ] Gentle Introduction to Adam Optimization [here](https://machinelearningmastery.com/adam-optimization-algorithm-for-deep-learning/).
- [x] A Gentle Introduction to GAN Loss Functions [here](https://machinelearningmastery.com/generative-adversarial-network-loss-functions/).
- [ ] How to choose Loss Functions in Neural Networks [here](https://machinelearningmastery.com/how-to-choose-loss-functions-when-training-deep-learning-neural-networks/).
- [x] How to implement Wasserstein Loss for GAN [here](https://machinelearningmastery.com/how-to-implement-wasserstein-loss-for-generative-adversarial-networks/).
- [ ] Understanding RMSprop [here](https://towardsdatascience.com/understanding-rmsprop-faster-neural-network-learning-62e116fcf29a).

**Comment by threeal** [comment]
Moved [here](https://gist.github.com/threeal/2a0a2999f7f0ef1d4e4a1dc3a95b1e13)
